src/lib/utils/tax.ts

```diff
@@ -5,32 +5,51 @@ import { GameState } from "features/game/types/game";
  * Returns the tax rate when withdrawing SFL
  * Smart contract uses a base rate of 1000 for decimal precision. 10% = 100
  */
-export function getTax({ amount, game }: { amount: Decimal; game: GameState }) {
-  // 10%
-  let tax = 100;
-
+function getTaxPercentage(amount: Decimal) {
   if (amount.lessThan(10)) {
-    // 30%
-    tax = 300;
-  } else if (amount.lessThan(100)) {
-    // 25%
-    tax = 250;
-  } else if (amount.lessThan(1000)) {
+    return 30;
+  }
+
+  if (amount.lessThan(100)) {
+    return 25;
+  }
+
+  if (amount.lessThan(1000)) {
     // 20%
-    tax = 200;
-  } else if (amount.lessThan(5000)) {
+    return 20;
+  }
+
+  if (amount.lessThan(5000)) {
     // 15%
-    tax = 150;
+    return 15;
   }
 
+  // 10%
+  return 10;
+}
+
+export function getTax({
+  amount,
+  game,
+}: {
+  amount: Decimal;
+  game: GameState;
+}): number {
+  amount = amount.sub(game.bank.taxFreeSFL);
+
+  if (amount.lte(0)) return 0;
+
+  let percentage = getTaxPercentage(amount);
+
   if (game.island.type !== "basic") {
-    tax -= 25; // 2.5% reduction
+    percentage -= 2.5; // 2.5%
```

Comment:
Is this going to become float or decimal?  You want to ensure decimal, right?

Reply:
I'm not too worried about float versus Decimal in this case, the business logic is handled on the backend when preparing this.

This util is mainly for UI purposes where there is small chance of funky behaviour (we typically cut off to 2 decimal places in displays though)

Reply:
@adamhannigan My `bank/taxfreeSFL` value bounces (increases/decreases) relative to "fixed point math" because of float issues with incrementing (via marketplace) or decreasing (via withdrawals).  So there are still (Dec 26 2026) issues at least with the back-end and perhaps with the front-end too.